tests/unittest_brain.py

```diff
@@ -2720,6 +2720,46 @@ def other_test(a, b, *, c=1):
             assert isinstance(inferred, astroid.Const)
             assert inferred.value == expected_value
 
+    def test_partial_assignment(self):
+        """Make sure partials are not assigned to original scope."""
+        ast_nodes = astroid.extract_node(
+            """
+        from functools import partial
+        def test(a, b): #@
+            return a + b
+        test2 = partial(test, 1)
+        test2 #@
+        def test3_scope(a):
+            test3 = partial(test, a)
+            test3 #@
+        """
+        )
+        func1, func2, func3 = ast_nodes
+        assert func1.parent.scope() == func2.parent.scope()
+        assert func1.parent.scope() != func3.parent.scope()
+        partial_func3 = next(func3.infer())
+        # use scope of parent, so that it doesn't just refer to self
+        scope = partial_func3.parent.scope()
+        assert scope.name == "test3_scope", "parented by closure"
+
+    def test_partial_does_not_affect_scope(self):
+        """Make sure partials are not automatically assigned."""
+        ast_nodes = astroid.extract_node(
+            """
+        from functools import partial
+        def test(a, b):
+            return a + b
+        def scope():
+            test2 = partial(test, 1)
+            test2 #@
+        """
+        )
+        test2 = next(ast_nodes.infer())
+        mod_scope = test2.root()
+        scope = test2.parent.scope()
+        assert set(mod_scope) == {"test", "scope", "partial"}
+        assert set(scope) == {"test2"}
```

Comment:
before, there was an extra item in locals: `{"test2": PartialFunction("test"), "test": PartialFunction("test")}`